0.3.0

MrSerge01 opened the pull request:
#39

Golem642 committed:
unmuting repaired

MrSerge01 committed:
Merge branch 'SokoraDesu:dev' into dev

iakzs committed:
trying pt.1

iakzs committed:
Merge branch 'Chiissu:dev' into dev

MrSerge01 committed:
Merge pull request #10 from iakzs/dev

sigma kz makes stuff for sigma sokowora

MrSerge01 committed:
the grand suffering pt1

MrSerge01 committed:
the grand suffering pt2

GoLittie committed:
settings command embed redesign

ZakaHaceCosas committed:
fix deprecation

ZakaHaceCosas committed:
very slightly restyle math command

ZakaHaceCosas committed:
Add support for server invites in serverboard

ZakaHaceCosas committed:
Replace underscore with whitespace when showing settings

ZakaHaceCosas committed:
Fix a few typos

ZakaHaceCosas committed:
Remove Instagram link (unused)

ZakaHaceCosas committed:
Humanize settings display + capitalize() don't return undefined

ZakaHaceCosas committed:
Format markdown + Add code style guidelines

ZakaHaceCosas committed:
Add timestamp and See profile to edit/remove mod embeds

ZakaHaceCosas committed:
i think this makes SQL more performant?

ZakaHaceCosas committed:
Allow to set serverboard invite channel

ZakaHaceCosas committed:
Prettify code

ZakaHaceCosas committed:
Return of roles param + don't prefer strict comparison

ZakaHaceCosas committed:
Simpler Enabled/Disable settings text

MrSerge01 committed:
Merge pull request #11 from ZakaHaceCosas/dev

some (possibly cool) stuff

ZakaHaceCosas committed:
Fix an error that made /serverboard crash Sokora

ZakaHaceCosas committed:
Merge branch 'Chiissu:dev' into dev

MrSerge01 committed:
Merge pull request #12 from ZakaHaceCosas/dev

Fix `/serverboard` crash

ZakaHaceCosas committed:
Format all code + Fix a few typos + All trailing commas

ZakaHaceCosas committed:
Merge branch 'Chiissu:dev' into dev

ZakaHaceCosas committed:
Update deps

ZakaHaceCosas committed:
Remove stupid type change

ZakaHaceCosas committed:
Disallow unused stuff + Remove Ephemeral flag import

MrSerge01 committed:
Merge pull request #13 from ZakaHaceCosas/dev

Some changes

ZakaHaceCosas committed:
SIKE!

ZakaHaceCosas committed:
Merge remote-tracking branch 'refs/remotes/origin/dev' into dev

ZakaHaceCosas committed:
THAT SHOULD NOT HAVE HAPPENED (undo commit)

ZakaHaceCosas committed:
guess whos dumb

MrSerge01 committed:
Merge pull request #14 from ZakaHaceCosas/dev

SIKE

ZakaHaceCosas committed:
Add security & verification checkbox to serverEmbed

ZakaHaceCosas committed:
Remove unused

ZakaHaceCosas committed:
Add (serverowner) and (currentdate) codes, and /help codes command

ZakaHaceCosas committed:
make help better or smth

ZakaHaceCosas committed:
Update (currentdate) code + Rename randomise => randomize (US eng)

ZakaHaceCosas committed:
Add (username) dynamic code

ZakaHaceCosas committed:
Support dynamic codes for news

ZakaHaceCosas committed:
Rename dynamic codes to dynamic variables

ZakaHaceCosas committed:
i forgor

ZakaHaceCosas committed:
Make security (well safety) string's code more logical

ZakaHaceCosas committed:
i forgor

MrSerge01 committed:
Merge pull request #15 from ZakaHaceCosas/dev

Show more info on serverEmbed + Add more (variables) + Add help command

MrSerge01 committed:
the grand suffering pt2.1

ZakaHaceCosas committed:
Revert a change that broke graph

MrSerge01 committed:
Merge pull request #16 from ZakaHaceCosas/dev

Revert a change that broke graph

ZakaHaceCosas committed:
a

ZakaHaceCosas committed:
Delete from public list servers that aren't public anymore

ZakaHaceCosas committed:
Merge branch 'Chiissu:dev' into dev

ZakaHaceCosas committed:
Now the ID is not hardcoded

ZakaHaceCosas committed:
Fun fact: GitHub is case-sensitive, apparently

ZakaHaceCosas committed:
yeah im stupid :sob:

ZakaHaceCosas committed:
Use genColor() for RPS

ZakaHaceCosas committed:
Better type randomize() and use it for RPS

ZakaHaceCosas committed:
Reduce RPS-code duplications

ZakaHaceCosas committed:
Add mention() and modActionEmbed()

ZakaHaceCosas committed:
Make modActionEmbed() reply by itself.

MrSerge01 committed:
Merge pull request #17 from ZakaHaceCosas/dev

Fix non-public-anymore servers staying in the list

ZakaHaceCosas committed:
Add meaning easter egg

MrSerge01 committed:
the grand suffering pt2.2

ZakaHaceCosas committed:
Merge branch 'Chiissu:dev' into dev

ZakaHaceCosas committed:
Improve typing of some utils

ZakaHaceCosas committed:
Rebuild and deploy-in-more-places modActionEmbed()

ZakaHaceCosas committed:
Update MD files because why not

MrSerge01 committed:
Merge pull request #18 from ZakaHaceCosas/dev

Improve source code overall + Add 'meaning' easter egg

MrSerge01 committed:
the grand suffering pt2.3

ZakaHaceCosas committed:
this should theoretically fix all known issues

MrSerge01 committed:
add guild/global command register commands

MrSerge01 committed:
easter eggs are stored in an array now

MrSerge01 committed:
logError pt1

MrSerge01 committed:
help me please

MrSerge01 committed:
suffering

MrSerge01 committed:
unfinished, pleaes someone do changelog else i yeet

ZakaHaceCosas committed:
Merge branch '0.2.1' into dev (hope nothing breaks)

ZakaHaceCosas committed:
minor tweaks

ZakaHaceCosas committed:
implement /changelog

ZakaHaceCosas committed:
fix no serverboard invites in rulesChannel-lessservers

ZakaHaceCosas committed:
Show ping in /about

ZakaHaceCosas committed:
properly get the path to CHANGELOG.md from /changelog

ZakaHaceCosas committed:
make changelog buttons work

ZakaHaceCosas committed:
minor updates

MrSerge01 committed:
a

ZakaHaceCosas committed:
b

ZakaHaceCosas committed:
correctly add .txt transcriptions to message logs

MrSerge01 committed:
get some utils up to date with 0.2.1

MrSerge01 committed:
rooftop, 52nd floor !!!

ZakaHaceCosas committed:
make starboard work (it didn't cause missing intent)

ZakaHaceCosas committed:
Update Sokora's changelog

ZakaHaceCosas committed:
make code formatting consistent

MrSerge01 committed:
hhhhhhhhh

MrSerge01 committed:
THE ERROR BOUNDARY IS DONE

iakzs committed:
forgor to commit this

iakzs committed:
hi

forgor to commit this

dimkauzh commented:
![IMG_20250403_220534_383.jpg](https://github.com/user-attachments/assets/ecadd5b7-6d2e-4c0d-a72b-cd3184da8d17)

ZakaHaceCosas committed:
minor changes

ZakaHaceCosas committed:
add some emojis to settings

ZakaHaceCosas committed:
minor fixes

userandaname committed:
best commit oat

userandaname committed:
THERE WE GOOOOO

userandaname committed:
coldest commit oat

userandaname committed:
fixed for ghostbusters

MrSerge01 committed:
Merge pull request #21 from userandaname/dev

hi vro

MrSerge01 committed:
errorEmbed revamp

ZakaHaceCosas committed:
replace console.error() with errorEmbed() in most places

ZakaHaceCosas committed:
simplify type definitions for settings

ZakaHaceCosas committed:
add coin flip, because why not

ZakaHaceCosas committed:
found it.

MrSerge01 committed:
list works a little bit more now

dimkauzh commented:
hi chat

dimkauzh commented:
The developers are slacking (jk jk)

MrSerge01 committed:
i am sokodead

MrSerge01 committed:
no more... no more LIST

MrSerge01 committed:
leave_channel works

ZakaHaceCosas committed:
fix mistake (missing getSetting() arg)

ZakaHaceCosas committed:
fix 'null' settings not being properly handled

iakzs committed:
Added something to only enable some specified Easter eggs

iakzs committed:
i forgot to delete this

iakzs committed:
done

iakzs committed:
e

iakzs committed:
i forgor

iakzs committed:
i forgor 2

MrSerge01 committed:
Merge pull request #23 from iakzs/dev

Enable Easter eggs thingy

ZakaHaceCosas committed:
make code follow codebase's formatting

ZakaHaceCosas committed:
add /danger reset command for settings

MrSerge01 committed:
wip user settings

Golem642 committed:
buncha fixes

Fixed hackban, unban, clear
Fixed embeds (and made DM modEmbed clearer)
Semi-moved hardcoded error logs forwarding channel id into env

MrSerge01 committed:
user settings work

ZakaHaceCosas committed:
damn.

MrSerge01 committed:
the errorEmbed faction captured console.error

MrSerge01 committed:
hiring a hitman for me

ZakaHaceCosas committed:
fix a bunch of errors + setup ESLint for better error detection

ZakaHaceCosas committed:
additional fix

ZakaHaceCosas committed:
a

Golem642 committed:
Something quick before i break something else

Golem642 committed:
Merge branch 'dev' of https://github.com/Chiissu/Sokora into dev

MrSerge01 committed:
unfinished embed changes

iakzs committed:
hi chart.js - kz

iakzs committed:
how did i add that

MrSerge01 committed:
Merge pull request #24 from iakzs/dev

hi chart.js

MrSerge01 committed:
embed stuff more or less done + rps upd

Golem642 committed:
guild audit logs

basic handler for audit log events

MrSerge01 committed:
audit log improvements + cleaning imports up

MrSerge01 committed:
audit log messageDelete works with normal one

userandaname committed:
ok so this is me when

dimkauzh commented:
sokocooking

MrSerge01 committed:
Merge pull request #25 from userandaname/dev

[feat] Add top.gg statistic posting

MrSerge01 committed:
bulkDelete works + iterable settings start

MrSerge01 committed:
iterables (cv2 rewrite next)

ZakaHaceCosas committed:
fix user settings + implement functional top.gg reminders

ZakaHaceCosas committed:
make topgg reminders actually good and optimized
goos' idea, gj

dimkauzh commented:
> uhhh I died

I can confirm

userandaname committed:
AUGGHHHHHH THE SELF-HOSTERS WOULDVE HATED US

MrSerge01 committed:
Merge pull request #26 from userandaname/dev

[fein] AUGGHHHHHH THE SELF-HOSTERS WOULDVE HATED US

MrSerge01 committed:
cv2 rewrite pt1

ZakaHaceCosas committed:
/moderation can now list all cases

ZakaHaceCosas committed:
/moderation can now filter by case type

ZakaHaceCosas committed:
add pagination to /moderation cases + minor fixes

ZakaHaceCosas committed:
(try to) avoid code duplication

ZakaHaceCosas committed:
prefer 'Mute' over 'MUTE'

ZakaHaceCosas committed:
done a bunch of changes
- added silent option and moderation.silent setting to make the modEmbed ephemeral (so the user doesn't realize)
- fixed /ban allowing to ban users not in the guild
- fixed mod commands overall not considering that the owner IS able to ban even if the target user has a higher role than him
- fixed /kick allowing to kick users not in the guild
- fixed /unmute not unmuting
- made code styling more consistent (try/catch instead of .catch())

ZakaHaceCosas committed:
add autocomplete to /danger reset

ZakaHaceCosas committed:
fix 'Unknown message' error when deleting messages with a collector

ZakaHaceCosas committed:
made welcome.roles work

MrSerge01 committed:
cv2 rewrite pt1.69420.1 (visibly a wip)

ZakaHaceCosas committed:
why?

ZakaHaceCosas committed:
cv2 rewrite pt1.69420.5 (UNFINISHED)
- this HALF works, HALF doesn't

MrSerge01 committed:
cv2 rewrite pt2 (more finished)

ZakaHaceCosas committed:
make the settings thingy gone when it times out

ZakaHaceCosas committed:
some code QL improvements

ZakaHaceCosas committed:
bye bye /changelog (kinda)

ZakaHaceCosas committed:
with CV2 it doesn't make sense to speak of IDs

ZakaHaceCosas committed:
remove unused settings

ZakaHaceCosas committed:
emojis :+1: :sunglasses: :skull: :fire:

MrSerge01 committed:
cv2 rewrite pt3 (more more finished)

ZakaHaceCosas committed:
now, if you star a replying message, replied guy also gets an embed

ZakaHaceCosas committed:
now easter eggs are ACTUALLY easter eggs (AKA they're rare)

MrSerge01 committed:
user settings cv2

userandaname committed:
LMFAO

userandaname committed:
add <meta> fetching

userandaname committed:
i can finally sleep peacefully

userandaname committed:
final touches

userandaname committed:
probs better to do this tbf

userandaname committed:
fix regex in the case of extra params

userandaname reviewed (approved):
adding 6,250 lines in one commit is CRAZY

MrSerge01 committed:
Merge pull request #27 from userandaname/dev

add media fetching for messageDelete & messageUpdate events

MrSerge01 committed:
newline

MrSerge01 committed:
reset in settings cv2 pt1

MrSerge01 committed:
reset in settings cv2 pt2

MrSerge01 committed:
reset in settings cv2 pt3

ZakaHaceCosas committed:
add some input validation to cv2settings

MrSerge01 committed:
change some stuff about input validation

MrSerge01 committed:
cases redo + modActionEmbed removed

ZakaHaceCosas committed:
sokonsistency (make all embeds consistent)

ZakaHaceCosas committed:
update changelog

MrSerge01 committed:
multi messagedelete pt1

ZakaHaceCosas committed:
multi messageDelete pt2
(broken af)

MrSerge01 committed:
multi messagedelete pt3

MrSerge01 committed:
multi messagedelete pt 3.1

MrSerge01 committed:
bro

ZakaHaceCosas committed:
multi messageDelete pt3.2
so close yet so far

userandaname reviewed (approved):
🤑

dimkauzh reviewed (approved):
Sweet

Golem642 commented on example.env:
```diff
@@ -1 +1,5 @@
-TOKEN="YOURTOKEN"
+TOKEN="YOUR_TOKEN"
+DEV_ERROR_CHANNEL_ID="1343140645132308532"
```
Maybe remove our channel ID lol

MrSerge01 replied:
oh my lord

Golem642 commented on src/commands/games/ascii.ts:
Should we count this as a game ?

MrSerge01 replied:
valid point, next commit will resolve 👍

MrSerge01 replied:
resolved

Golem642 commented on src/commands/games/coin.ts:
```diff
@@ -0,0 +1,26 @@
+      iconURL: avatar,
+    })
+    .setDescription(
+      Math.random() >= 0.5 ? "The coin landed on **tails**!" : "The coin landed on **heads**!",
```
```JS
`The coin landed on **${Math.random >= 0.5 ? 'tails' : 'head'}**!`,
```

MrSerge01 replied:
thank you

Golem642 commented on src/commands/leaderboard.ts:
```diff
@@ -0,0 +1,111 @@
+  const row = new ActionRowBuilder<ButtonBuilder>().addComponents(
+    new ButtonBuilder()
+      .setCustomId("left")
+      .setEmoji("1298708251256291379")
```
Maybe replace magic numbers with some sort of config ?

MrSerge01 replied:
I mean, it kinda says that it's an emoji due to `.setEmoji`... so I think it's pretty much known that it's an ID

Golem642 replied:
Yeah but like we have a defined set of emojis, so why not replace the magic numbers by some actual def ?
Also this would be better for selfhosters

MrSerge01 replied:
you're right, i'll make a new thing in constants rq

Golem642 commented on src/commands/math/calc.ts:
```diff
@@ -0,0 +1,43 @@
+
+  let result: number;
+  try {
+    result = math.evaluate(expr);
```
Probably needs a timeout if for some reason a kid decides to send an expression that would take years to complete idk

MrSerge01 replied:
lmfao valid

ZakaHaceCosas replied:
expressions either can be done quickly or make the math library itself throw an error, so technically this already times out

MrSerge01 replied:
upd: zakaballer did the thing

Golem642 commented on src/commands/moderation/ban.ts:
```diff
@@ -0,0 +1,77 @@
+
+  try {
+    await Promise.all([
+      modEmbed(
```
iirc the DM when performing an action on someone works now right ?

MrSerge01 replied:
I hope so

MrSerge01 replied:
tested it and it turned out that it does not work, me and zaka fixed it, thank you :D

Golem642 commented on src/commands/moderation/warn.ts:
```diff
@@ -0,0 +1,49 @@
+      .setDescription(
+        "Inform the warned user of the moderator that took the action. Defaults to false.",
+      ),
+  )
```
We could use this in some other places

MrSerge01 replied:
true

MrSerge01 replied:
update: we do not need to use this in any place due to `silent` hiding the entire embed

Golem642 commented on src/commands/moderation/Note.ts:
Why remove note entirely ?

MrSerge01 replied:
it's basically the same thing as a discord message 😭

Golem642 commented on src/commands/user/info.ts:
```diff
@@ -0,0 +1,162 @@
+    (await genImageColor(undefined, avatar)) ??
+    genColor(200);
+
+  const embed = new EmbedBuilder()
+    .setAuthor({
+      name: `${dotCheck({ string: avatar, doubleSpace: true })}${target?.nickname ?? user.displayName}`,
+      iconURL: avatar,
+    })
+    .setFields({
+      name: `<:discord:1266797021126459423> • Discord info`,
+      value: [
+        `Username is **${user.username}**`,
+        `Display name is ${
+          user.displayName == user.username ? "*not there*" : `**${user.displayName}**`
+        }`,
+        `Created on **<t:${Math.round(user.createdAt.valueOf() / 1000)}:D>**`,
+      ].join("\n"),
+    })
+    .setFooter({ text: `User ID: ${user.id}` })
+    .setColor(embedColor);
+
```
We could have a separate embed file for this just like server embed, if we ever need to use it for moderation commands for example

MrSerge01 replied:
we're throwing this codebase away after 0.3, I don't think that we'd need a separate user embed

Golem642 commented on src/events/easterEggs/sokodunk.ts:
```diff
@@ -0,0 +1,22 @@
+    return;
+
+  await (message.channel as TextChannel).send(
+    "https://tenor.com/view/sokora-dunk-ice-skate-ice-dunk-balling-gif-7665972654807661282?quality=lossless",
```
I should polish the second one and publish it as well

Golem642 commented on src/events/messageDelete.ts:
```diff
@@ -1,28 +1,201 @@
+    if (message.partial) return;
+    const author = message.author;
+    if (!author)
+      return await errorEmbed({
+        client,
+        title: "Cannot log deleted message.",
+        reason: `Message ${message} lacks an author.`,
```
We technically could still log it

MrSerge01 replied:
true, but it's ensuring that the author exists in the first place (I don't know how a message can exist without the author...)

Golem642 commented on src/utils/database/disabledCommands.ts:
?

MrSerge01 replied:
will get removed, thanks

MrSerge01 replied:
oh wait it is removed LMAO, if we need to replace that (we won't, for now) we can just make an iterable setting

Golem642 commented on src/utils/embeds/modEmbed.ts:
```diff
@@ -1,194 +1,251 @@
+      });
   }
-
-  if (outsideError)
-    if (
-      !(await guild.members
-        .fetch(user.id)
-        .then(() => true)
-        .catch(() => false))
-    )
-      return await errorEmbed(
-        interaction,
-        `You can't ${action.toLowerCase()} ${name}.`,
-        "This user isn't in this server."
-      );
```
I don't remember why we removed this, it should default to true except for things like ban

MrSerge01 replied:
it isn't removed:
```ts
  if (outsideError)
    if (!guild.members.cache.has(user.id)) {
      const isBanError = (await guild.bans.fetch()).has(user.id) && action == "Ban";

      return await errorEmbed({
        interaction,
        title: `You can't ${action.toLowerCase()} ${name}.`,
        reason: isBanError ? "This user was already banned." : "This user isn't in this server.",
      });
    }
```

Golem642 reviewed (changes requested):
I know I'm annoying lol, I couldn't even work that much on this version so i'm trying to repay myself by reviewing this somewhat correctly
Might have missed a few things (I'm doing this on my phone, my laptop is still dead), overall really good stuff with a few questions and remarks
Good job guys

ZakaHaceCosas committed:
minor change

MrSerge01 committed:
issues are addressed

MrSerge01 committed:
guh

ZakaHaceCosas committed:
remove messageDelete grouping (we'll do it; later) + remove duplicate code

ZakaHaceCosas committed:
fixed the logo
now it's perfect.

MrSerge01 committed:
smol.

ZakaHaceCosas committed:
who messed this up

MrSerge01 committed:
if any file doesn't come through, i will BOIL git

ZakaHaceCosas committed:
permission error fix

ZakaHaceCosas committed:
various fixes n tweaks
- fixed news edit failing because news title was too long
- fixed starboard theoretically allowing voice channels and stuff
- added channel check so if, e.g., a channel sokora uses for smth is deleted, it's properly handled
- tweaked RPS embed
- tweaked user info, now there's no buttons, everything shows at once

MrSerge01 committed:
changes

ZakaHaceCosas committed:
remove ascii (flawfull)

ZakaHaceCosas committed:
Update rps.ts

MrSerge01 committed:
changes

MrSerge01 committed:
ultimate sub 30 command

MrSerge01 committed:
after a quick review,

MrSerge01 committed:
suffering

ZakaHaceCosas committed:
fixed the suffering :D

MrSerge01 committed:
some additional bug fixes :D

ZakaHaceCosas committed:
minor tweaks

ZakaHaceCosas reviewed (approved):
Reviewed everything and it all looks great now. Congratulations y'all as this is actually a great release.
@Golem642 your review is needed for GitHub to mark this as mergeable, thank you.

LGTM.

Golem642 commented on src/utils/embeds/modEmbed.ts:
```diff
@@ -1,194 +1,247 @@
             .setAuthor({
-              name: `•  You got ${action.toLowerCase()}.`,
-              iconURL: user.displayAvatarURL()
+              name: `${dotCheck({ string: serverAvatar, doubleSpace: true })}${customText?.dmTitle ?? `You got ${action?.toLowerCase()} from ${guild.name}`}`,
```
Why not have the server name as the embed name and the "You got ..." as the embed title ?

MrSerge01 replied:
that's the DM embed, so the avatar is from the guild to make sure the user knows exactly where it's coming from

Golem642 commented on src/utils/embeds/modEmbed.ts:
```diff
@@ -1,194 +1,247 @@
+      name: `${dotCheck({ string: avatar, doubleSpace: true })}${customText?.logTitle ?? author}`,
+      iconURL: avatar,
+    })
     .setDescription(generalValues.join("\n"))
```
If you removed the showModerator option, maybe also remove the moderator from the DMed embed ? To avoid targeting

MrSerge01 replied:
as I said, the `silent` option can be used to hide the whole embed. + if a user targets a person, won't they simply get banned lmao

userandaname replied:
> if a user targets a person, won't they simply get banned lmao

fair, but @Golem642 might assume that the moderator could get harassed from alternate accounts, or smth along those lines. the `silent` option should cover things though, so should be fine imo

Golem642 reviewed (changes requested):
We need to test this a bit further imo, some changes I'm not so sure about but i mean if you guys want we could also send it into the wild and see how people break it

MrSerge01 committed:
mostly perm stuff

MrSerge01 committed:
oh shoot

MrSerge01 committed:
no more subcommand group handling

ZakaHaceCosas committed:
newline fix

MrSerge01 committed:
last push pt1

MrSerge01 committed:
last push pt2

ZakaHaceCosas committed:
last push pt2.1

ZakaHaceCosas committed:
last push pt2.1 DLC 1

MrSerge01 committed:
last push pt2.1 DLC 2

ZakaHaceCosas committed:
last push pt2.1 Expansion Pack 1

MrSerge01 committed:
resurrection of subcommand groups (why)

Golem642 commented on src/utils/multiReact.ts:
Still sad about this btw

Golem642 commented on src/utils/database/settings.ts:
```diff
@@ -1,200 +1,367 @@
-export function listPublicServers() {
-  return (listPublicQuery.all() as TypeOfDefinition<typeof tableDefinition>[]).map(entry =>
-    JSON.parse(entry.guildID)
+export function listPublicServers(): Promise<
+  {
+    guildID: string;
+    showInvite: boolean;
+    inviteChannelId: string | null;
+  }[]
+> {
+  const publicGuildSet = new Set(
+    (listPublicQuery.all() as TypeOfDefinition<typeof tableDefinition>[]).map(entry =>
+      JSON.parse(entry.guildID),
+    ),
+  );
+
+  const inviteGuildsSet = new Set(
+    (listPublicWithInvitesEnabledQuery.all() as TypeOfDefinition<typeof tableDefinition>[]).map(
+      entry => JSON.parse(entry.guildID),
+    ),
+  );
+
+  return Promise.all(
+    Array.from(publicGuildSet).map(async entry => {
+      const inviteChannel = await getSetting(entry, "serverboard", "invite_channel");
+      return {
+        guildID: entry,
+        showInvite: inviteGuildsSet.has(entry),
+        inviteChannelId: inviteChannel ? inviteChannel.toString() : null,
+      };
+    }),
   );
 }
```
Tested and confirmed working ?

MrSerge01 replied:
we had a major vulnerability in there, but we resolved it obviously
it works properly now

ZakaHaceCosas replied:
yep

Golem642 reviewed (commented):
If the serverboard invites works now and doesn't break anything you may have my validation

Golem642 reviewed (approved):
Alright alright. Here, you all deserved it :

LGTM

ZakaHaceCosas commented:
HOLY SOKORA THANK YOU MAN!

ZakaHaceCosas committed:
the final commit?
re-adds multiReact under request, nothing else

ZakaHaceCosas committed:
there ALWAYS has to be a last minute fix doesn't it?

proJM-Coding commented:
HONK

proJM-Coding commented:
Nice to see you again @MrSerge01 :)
